src/rules/noNamespaceRule.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "'namespace' and 'module' are disallowed";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NullWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class NullWalker extends Lint.RuleWalker {
+    public visitNode(node: ts.Node) {
+        super.visitNode(node);
+        if (node.kind !== ts.SyntaxKind.ModuleDeclaration) { return; }
+        let decl = <ts.ModuleDeclaration>node;
+        // declare module 'foo' {} is an external module, not a namespace.
+        if (decl.name.kind === ts.SyntaxKind.StringLiteral) { return; }
+        /* tslint:disable:no-bitwise */
+        if (node.flags & ts.NodeFlags.Ambient && this.hasOption("allow-declarations")) { return; }
```

Comment:
Can you wrap `node.flags & ts.NodeFlags.Ambient` in parens? This code is logically correct as is, I just always forget bitwise operator precedence, so it's nice to have it visually distinguished